DataFormats/TrackerCommon/interface/TrackerTopology.h

```diff
@@ -102,6 +102,22 @@ class TrackerTopology {
     unsigned int sterMask_;
   };
 
+  enum DetIdFields {
+    PBModule, PBLadder, PBLayer,
+    PFModule, PFPanel, PFBlade, PFDisk, PFSide, 
+    /* TODO: this can be extended for all subdetectors */
+    DETID_FIELDS_MAX
+  };
+
+ private:
+  struct BitmaskAndSubdet { 
+    unsigned int const& startBit; 
+    unsigned int const& mask;
+    int subdet;
+  };
+  const BitmaskAndSubdet bits_per_field[DETID_FIELDS_MAX];
```

Comment:
I still see ``` unsigned int const& startBit ```
The pfVals and pbvals are not modified in TT implementation and can't be from outside methods. So, it's as good as const.
Performance-wise, it would be better to use ``` unsigned int startBit ``` (can be const FWIW).